README.md

```diff
@@ -1,25 +1,109 @@
 # Ember-cli-deploy-redis
 
-This README outlines the details of collaborating on this Ember addon.
+> An ember-cli-deploy-plugin to upload files to a Redis store
+
+**This plugin is only compatible with ember-cli-deploy versions >= 0.5.0**\
+
+This plugin uploads one or more files to a specified Redis store.
+
+More often than not this plugin will be used in conjunction with the [lightening method of deployment][1] where the ember application assets will be served from S3 and the index.html file will be served from Redis. However, it can be used to upload any files to a Redis store.
+
+## What is an ember-cli-deploy plugin?
+
+A plugin is an addon that can be executed as a part of the ember-cli-deploy pipeline. A plugin will implement one or more of the ember-cli-deploy's pipeline hooks.
+
+For more information on what plugins are and how they work, please refer to the [Plugin Documentation][2]
 
 ## Installation
+Run the following command in your terminal:
 
-* `git clone` this repository
-* `npm install`
-* `bower install`
+```bash
+ember install ember-cli-deploy-redis
+```
 
-## Running
+## ember-cli-deploy hooks implemented
+- `configure`
+- `upload`
+- `activate`
+- `didDeploy`
 
-* `ember server`
-* Visit your app at http://localhost:4200.
+## Configuration options
 
-## Running Tests
+### host
+
+The Redis host. If [url](#url) is defined, then this option is not needed.
+
+*Default:* `'localhost'`
+
+### port
+
+The Redis port. If [url](#url) is defined, then this option is not needed.
+
+*Default:* `6379`
+
+### url
+
+A Redis connection url to the Redis store
+
+*Example:* 'redis://some-user:some-password@some-host.com:1234'
+
+### filePattern
+
+Files matching this pattern will be uploaded to Redis. By default this option will use the `distFiles` property of the deployment context.
 
-* `ember test`
-* `ember test --server`
+*Default:* `context.distFiles + '/index.html'`
```

Comment:
This should probably be `distDir`, but what's more, this value should probably just be `"index.html"` and be applied to files in `distDir`, which would be it's own configuration option. Is there a use case for deploying something that is not under `distDir`?